apm-agent-plugins/apm-ratpack-plugin/src/main/java/co/elastic/apm/agent/ratpack/AbstractExecStarterInstrumentation.java

```diff
@@ -0,0 +1,288 @@
+                            public void accept(Transaction transaction) {
+
+                                execution.maybeGet(Context.TYPE)
+                                    .ifPresent(new FillTransactionContext(transaction, webConfiguration));
+
+                                transaction.end();
+                            }
+                        });
+                    }
+                });
+        }
+
+        private static class FillTransactionContext implements Consumer<Context> {
+
+            private final Transaction transaction;
+            private final WebConfiguration webConfiguration;
+
+            public FillTransactionContext(Transaction transaction, WebConfiguration webConfiguration) {
+                this.transaction = transaction;
+                this.webConfiguration = webConfiguration;
+            }
+
+            @Override
+            public void accept(Context context) {
+
+                transaction.withResult(context.getResponse().getStatus().toString());
+
+                fillTransactionName(transaction, context);
+                fillRequestContext(transaction.getContext().getRequest(), context);
+                fillResponseContext(transaction.getContext().getResponse(), context);
+            }
+
+            private void fillTransactionName(Transaction transaction, Context context) {
+
+                transaction.getName()
+                    .append(context.getRequest().getMethod().getName())
+                    .append(" /")
+                    .append(context.getRequest().getPath());
+            }
+
+            private void fillRequestContext(Request request, Context context) {
+
+                if (transaction.isSampled() && webConfiguration.isCaptureHeaders()) {
+
+                    for (Cookie cookie : context.getRequest().getCookies()) {
+                        request.addCookie(cookie.name(), cookie.value());
+                    }
+
+                    for (final String name : context.getRequest().getHeaders().getNames()) {
+                        for (final String value : context.getRequest().getHeaders().getAll(name)) {
+                            request.addHeader(name, value);
+                        }
+                    }
+                }
+
+                final ServerConfig serverConfig = context.getServerConfig();
+
+                request.withHttpVersion(getHttpVersion(context.getRequest().getProtocol()));
+                request.withMethod(context.getRequest().getMethod().getName());
+
+                request.getSocket()
+                    .withRemoteAddress(ClientIpUtils.getRealIp(request.getHeaders(), context.getRequest().getRemoteAddress().toString()));
+
+                request.getUrl()
+                    .withPort(serverConfig.getPort())
+                    .withPathname(context.getRequest().getUri())
+                    .withSearch(context.getRequest().getQuery());
+
+                request.redactBody();
```

Comment:
add a TODO to record the body according to `capture_body` and `capture_body_content_types`